airflow/providers/google/cloud/transfers/postgres_to_gcs.py

```diff
@@ -41,26 +41,22 @@ class _PostgresServerSideCursorDecorator:
 
     def __init__(self, cursor):
         self.cursor = cursor
-        self.rows = []
         self.initialized = False
 
     def __iter__(self):
         return self
 
     def __next__(self):
-        if self.rows:
-            return self.rows.pop()
-        else:
-            self.initialized = True
-            return next(self.cursor)
+        self.initialized = True
+        return next(self.cursor)
 
     @property
     def description(self):
         """Fetch first row to initialize cursor description when using server side cursor."""
         if not self.initialized:
-            element = self.cursor.fetchone()
-            self.rows.append(element)
             self.initialized = True
+            self.cursor.fetchone()
+            self.cursor.scroll(0, mode='absolute')
```

Comment:
If `.scroll(-1)` is more clear then I change

Reply:
I like your current approach better.

Reply:
Sorry for the misleading.

First of all server-side cursor doesn't support `relative` mode by default. It should be initialized like `conn.cursor(name='...', scrollable=True). But absolute mode could be used.

But both scrolls cause a reset of `.description` attribute. So, unfortunately, they could not be used.